### ISSUE TITLE
Sanitize challenge and CTF names

### ISSUE THREAD

**Issue by Kileak** [OpenToAllCTF/OTA-Challenge-Bot#46]
- We need to check for # in challenge names (if people try to add a challenge and pass the channel name instead, the channel won't be created, but the challenge will, thus it's kinda broken)

**Comment by Grazfather** [comment]
Plus we need to sanitize other special chars

**Comment by Corb3nik** [comment]
I suggest maybe adding length limitations for CTF names as well as challenge names.
Using long challenge names complicate the usage a bit IMO.

Channels have a [max length of 21](https://get.slack.help/hc/en-us/articles/201654073-Rename-a-channel).
Maybe we can use : 
 - CTF max length of 10
 - Challenge max length of 10

This will force people to use short descriptive names instead of long ones that are hard to write down.

**Comment by Grazfather** [comment]
That sounds good (+1 char for the dash between).

For the challenge name: alpha + number + dash + underscore.

```python
import re
...
if re.match('^[\w-_]+$', chan_name) is not None
```

**Comment by wbowling** [comment]
Should abort the command or just just strip the bad characters?

**Comment by Kileak** [comment]
I think it should abort it, and show a message describing, that something bad was entered (We could use the description for a command for that, since it will be shown only in error case, so we could also add the limits there and raise an exception (or just show the single command usage)). 

By only stripping chars, we might end up with bad channel names anyways.

**Comment by Corb3nik** [comment]
Added length limits for CTF/challenge names. All that remains is whitelisting characters.

**Comment by Grazfather** [comment]
I'll do this tomorrow or tonight after work.

**Comment by Grazfather** [comment]
Fixed in #105
